ejb3/src/main/java/org/jboss/as/ejb3/component/stateful/StatefulSessionSynchronizationInterceptor.java

```diff
@@ -139,6 +154,22 @@ public Object processInvocation(final InterceptorContext context) throws Excepti
                         instance.getComponent().getCache().release(instance);
                     }
                 }
+                for(;;) {
+                    int state = invocationSyncState.get();
+                    if(state == SYNC_STATE_INVOCATION_IN_PROGRESS && invocationSyncState.compareAndSet(SYNC_STATE_INVOCATION_IN_PROGRESS, SYNC_STATE_NO_INVOCATION)) {
+                        break;
+                    } else if (state == SYNC_STATE_AFTER_COMPLETE_DELAYED) {
+                        try {
+                            //we know the transaction did not commit, as we are still active in the tx, this only happens through timeout
```

Comment:
Another possibility is that the transaction was manually acquired through some user code or framework and committed this way.  I think it's at least more consistent to track the transaction outcome and report it, even if it is very likely always to be false.